quinn-proto/src/cid_generator.rs

```diff
@@ -31,6 +31,33 @@ pub trait ConnectionIdGenerator: Send + Sync {
     fn cid_lifetime(&self) -> Option<Duration>;
 }
 
+/// Validates connection id in short header packet
+pub trait ConnectionIdValidator {
```

Comment:
The trait should be defined below the implementator.

Reply:
`ConnectionIdParser` was introduced in `shred.rs` instead of this, I'm not sure whether the new define order is what you want.